### ISSUE TITLE
Button Component WCAG Accessibility issues

### ISSUE THREAD

**Issue by chris-gds** [jpmorganchase/salt-ds#4312]
### Latest version

- [X] I have tested the latest version

### 1) Issue: Uppercase text 

<img width="810" alt="Salt Design System Button Component" src="https://github.com/user-attachments/assets/a691a904-b95f-4322-99fc-5094532ee816">

WCAG 1.4.12 Uppercase text decreases readability for all users, but especially for those with dyslexia or other cognitive disabilities. All-uppercase text reduces the distinguishable shapes of letters (since uppercase letters have uniform heights), making it harder to recognise words by their shapes. [1] [2]

WCAG 1.4.8 emphasises that text should be presented in ways that improve readability, including limiting the use of fully uppercase text.

[1] https://accessibility.huit.harvard.edu/design-readability#:~:text=Avoid%20changing%20the%20typeface%20from,Don't%20underline%20text.

[2] https://www.gov.uk/guidance/publishing-accessible-documents

 `--salt-text-action-textTransform: uppercase;`
[packages/theme/css/characteristics/text.css](https://github.com/jpmorganchase/salt-ds/blob/7d0d99890931ff09f725f324f731d44add56c52e/packages/theme/css/characteristics/text.css#L12)

### 2) Issue WCAG 2.5.5 Target Size (AAA)

Impact: Although this criterion is AAA, it’s worth noting that WCAG 2.5.5 recommends that touch targets (like buttons) be large enough (44 by 44 CSS pixels) to accommodate users who have motor impairments. A small button with uppercase text might be hard to tap or click accurately, especially on mobile devices.

### 3) Issue Disabled buttons

https://axesslab.com/disabled-buttons-suck/

### 4) Issue inconsistent focus styles

WCAG 2.4.7 Inconsistent focus styles (e.g., different elements having varying or missing focus indicators) can confuse users, especially keyboard and screen reader users who rely heavily on visual cues to navigate through content.

### Steps to reproduce

https://www.saltdesignsystem.com/salt/components/button/examples

### Expected behavior

1) Text to be `Sentence case`
2) All buttons to have a minimum size
3) More considerations for disabled buttons
4) Focus styles to be consistent

### Package name(s)

Core (@salt-ds/core), Theme (@salt-ds/theme)

### Browser

- [X] Chrome
- [X] Safari
- [X] Firefox
- [X] Microsoft Edge

### Operating system

- [X] macOS
- [X] Windows
- [X] Linux
- [X] iOS
- [X] Android

### Are you a JPMorgan Chase & Co. employee?

- [ ] I am an employee of JPMorgan Chase & Co.

**Comment by origami-z** [comment]
> 1) Issue: Uppercase text

We have received feedback around this. It's upper case to help distinguish transparent buttons with other elements (e.g. v.s. Text, v.s. Link). Do you have any alternative way to solve this?

Button as a component, is not the main content on the screen, so is "limited" in a way?

> 2) Issue WCAG 2.5.5 Target Size (AAA)

When using Touch density, Button meets 44px requirement

> 3) Issue Disabled buttons

What's your suggestion? 

[Focusable when disabled](https://www.saltdesignsystem.com/salt/components/button/examples#focusable-when-disabled) is one way for us to help the situation.

> 4) Issue inconsistent focus styles

All Salt components has consistent styling (dotted blue), it's a bug on the website itself which doesn't use Salt Link.

cc @jake-costa

**Comment by jake-costa** [comment]
@chris-gds First, I want to thank you for bringing this up and focusing on digital accessibility. It’s great to have users like you considering accessibility from the start, ensuring an inclusive and accessible experience for everyone. That said, I’d like to clarify that Salt currently follows [WCAG 2.1 AA](https://www.w3.org/TR/WCAG21/) created by the [W3C](https://www.w3.org/) as the accessibility standard. Any other third-party guidelines not associated with the [W3C](https://www.w3.org/) would be out of scope.

> 1) Issue: Uppercase text

As mentioned "It's upper case to help distinguish transparent buttons with other elements (e.g. v.s. Text, v.s. Link)"

Regarding [WCAG 1.4.12](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html), it focuses on ensuring users can adjust text spacing properties—such as line height, letter spacing, and word spacing—but it doesn’t directly restrict the use of uppercase text. 

As for [WCAG 1.4.8](https://www.w3.org/WAI/WCAG21/Understanding/visual-presentation.html), it falls under AAA standards, which are out of scope for us. Additionally, it doesn’t directly restrict the use of uppercase text either

The Harvard accessibility link you provided isn’t affiliated with the [W3C](https://www.w3.org/). In addition, while it does **recommend** avoiding the use of all-uppercase text, this is more of a recommendation rather than a strict requirement. Similarly, the gov.uk link is also not affiliated with the [W3C](https://www.w3.org/) and primarily provides guidance on digital documents rather than web applications with interactive components.

> 2) Issue WCAG 2.5.5 Target Size (AAA)

As you noted, [WCAG 2.5.5](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) is a AAA criterion, therefore it is out of scope for our current standards and isn’t considered an accessibility violation. 

> 3) Issue Disabled buttons

While this is certainly a hot topic within the accessibility community, there’s currently no restriction in [WCAG 2.1 AA](https://www.w3.org/TR/WCAG21/) that prevents the use of disabled buttons in applications. Therefore, it isn’t considered an accessibility violation.

> 4) Issue inconsistent focus styles

Thanks for bringing this up! As mentioned, all Salt components are designed with consistent focus styling. This particular issue seems to be related to the site implementation rather than the component itself.

**Comment by chris-gds** [comment]
> We have received feedback around this. It's upper case to help distinguish transparent buttons with other elements (e.g. v.s. Text, v.s. Link). Do you have any alternative way to solve this?

> As mentioned "It's upper case to help distinguish transparent buttons with other elements (e.g. v.s. Text, v.s. Link)"

Should the button look like text due the Design, this probably needs a rethink so it looks like a `button` - as it _is_ a `button`.  This would impact [information and relationships 1.3.1.](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)

> Regarding [WCAG 1.4.12](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html), it focuses on ensuring users can adjust text spacing properties—such as line height, letter spacing, and word spacing—but it doesn’t directly restrict the use of uppercase text.

> As for [WCAG 1.4.8](https://www.w3.org/WAI/WCAG21/Understanding/visual-presentation.html), it falls under AAA standards, which are out of scope for us. Additionally, it doesn’t directly restrict the use of uppercase text either

> The Harvard accessibility link you provided isn’t affiliated with the [W3C](https://www.w3.org/). In addition, while it does recommend avoiding the use of all-uppercase text, this is not a strict requirement. Similarly, the gov.uk link is also not affiliated with the [W3C](https://www.w3.org/) and primarily provides guidance on digital documents rather than web applications with interactive components.

There are not any strong arguments to continue to use UPPERCASE text compared to Sentence text. Evidence demonstrates that's it's more readable especially for readers with cognitive issues like [dyslexia](https://www.bdadyslexia.org.uk/advice/employers/creating-a-dyslexia-friendly-workplace/dyslexia-friendly-style-guide#:~:text=Avoid%20underlining%20and%20italics%20as,letters%20are%20easier%20to%20read.).  

Sometimes people are excluded on purpose.

> What's your suggestion?

One solution is to remove `buttons` that are disabled entirely - why does it need to be there?

> All Salt components has consistent styling (dotted blue), it's a bug on the website itself which doesn't use Salt Link.

The styling seems to change depending on the colour of the button rather than be consistent

**Comment by jake-costa** [comment]
>Bryan Gill, JPMorgan Chase's head of the Global Office of Disability Inclusion and the bank's head of Global Neurodiversity writes "[...when we dismiss or minimize the neurodiverse, we also reinforce the notion that difference equals weakness—and that those who are different are somehow insufficient. ](https://www.jpmorganchase.com/newsroom/stories/neurodiversity)"

I want to emphasize that our use of transparent buttons or uppercase text in certain UI elements, and our focus on WCAG 2.1 AA rather than AAA compliance, absolutely does not mean we are dismissing or minimizing the needs of neurodiverse users or those with cognitive disabilities.

>There are not any strong arguments to continue to use UPPERCASE text compared to Sentence text. Evidence demonstrates that's it's more readable, especially for readers with cognitive issues like [dyslexia](https://www.bdadyslexia.org.uk/advice/employers/creating-a-dyslexia-friendly-workplace/dyslexia-friendly-style-guide#:~:text=Avoid%20underlining%20and%20italics%20as,letters%20are%20easier%20to%20read.).

>Should the button look like text due the Design, this probably needs a rethink so it looks like a button - as it is a button. This would impact [information and relationships 1.3.1.](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)

I encourage you to look at [W3C Non-Text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)
_"A button which has a distinguishing indicator such as position, **text style**, or context does not need a contrasting visual indicator to show that it is a button..."_

To be clear:

1. Accessibility is a core priority for us. We strive to create inclusive designs for all users, including those with diverse cognitive needs.

2. WCAG 2.1 AA compliance, which we target, is a robust standard that addresses many accessibility needs. The [W3C Conformance Requirements](https://www.w3.org/WAI/WCAG21/Understanding/conformance) itself notes that AAA compliance is not always feasible or necessary for all content.

_"It is not recommended that Level AAA conformance be required as a general policy for entire sites because it is not possible to satisfy all Level AAA Success Criteria for some content."_

3. We continuously test and refine our components to ensure they work for as many people as possible.

>One solution is to remove buttons that are disabled entirely - why does it need to be there? It can be brought back once it's active.

When to use disabled as a function depends on context, audience, expectations, and more. Often there are better ways to handle making a feature inactive, but a blanket ban for a design system is not ideal at this time. That said, we take all accessibility feedback very seriously and use it to improve our components. I want to assure you that we will review your feedback internally. Our team is committed to ongoing evaluation and improvement of our accessibility practices.

Lastly, being a design system, it's crucial to distinguish between design choices that may require thoughtful implementation and accessibility violations that actively dismiss or minimize the needs of any user group. Our approach aims to balance accessibility, usability, and design considerations for a wide range of users.

We appreciate you advocating for neurodiverse users. If you have specific examples where our design system is causing accessibility violations, we'd be very interested in hearing more to inform our ongoing improvements.

**Comment by chris-gds** [comment]
> I want to emphasize that our use of transparent buttons or uppercase text in certain UI elements, and our focus on WCAG 2.1 AA 

A transparent button that looks like text would be a fail of WCAG 2.1 **A** under 1.3.1 as it's not visually aligned to the underlying code.

> not mean we are dismissing or minimizing the needs of neurodiverse users or those with cognitive disabilities.

To confirm, there is a continued use of UPPERCASE text even when you're aware users with cognitive disabilities face issues with this yet you're confident this is not _"dismissing or minimizing the needs of neurodiverse users or those with cognitive disabilities."_

Thanks for your time @jake-costa.

**Comment by joshwooding** [comment]
Thanks for the feedback, we are all passionate about accessibility and it sounds like some of that passion is boiling over. Due to the direction this conversation might go in, I'm going to lock this issue. Have a nice day.
